### PR TITLE
1295 laos eoc forecast weather map overlays

### PR DESCRIPTION
**Pull request by IgorNadj**
### Changes:
- Add Laos EOC map overlays for forecast data
- Change WeatherService to always require start/end dates (and so make sure that forecast weather requests that specify start/end dates are handled correctly)
---

### Screenshots:
https://github.com/beyondessential/tupaia-backlog/issues/1295#issuecomment-705946095

### CODE REVIEW

File: packages/data-broker/src/services/weather/WeatherService.js [hunk 1]
```diff
@@ -149,39 +143,55 @@ export class WeatherService extends Service {
...
+        data: filteredData,
+      };
+    };
+
     // Run requests in parallel for performance
     const getDataForEntity = async entity => {
       const { lat, lon } = entity.pointLatLon();
 
-      const days = 1; // just need today
+      // Maximum forecast is 16 days, we request all of it and filter it dow to the dates we need.
```
Comment (edmofro) [inline packages/data-broker/src/services/weather/WeatherService.js:186]:
```suggestion
      // Maximum forecast is 16 days, we request all of it and filter it down to the dates we need.
```

File: packages/data-broker/src/tests/services/weather/WeatherService.test.js [hunk 2]
```diff
@@ -136,10 +136,30 @@ describe('WeatherService', () => {
...
     });
+
+    it('throws when no dates are provided', async () => {
+      const mockModels = await createMockModelsStubWithMockEntity();
+
+      const mockApi = createWeatherApiStubWithMockResponse();
+
+      const service = new WeatherService(mockModels, mockApi);
+
+      const functionCall = async () =>
```
Comment (edmofro) [inline packages/data-broker/src/tests/services/weather/WeatherService.test.js:147]:
```suggestion
      const pullWithUndefinedDates = async () =>
```
With `functionCall` I have to think a bit harder about "ok, but what function call?"

**Review by edmofro — APPROVED** [review]
Nice! A very clean implementation, shows all of your work is paying off :-)

I've made a couple of very minor comments since there was nothing else to pick up, feel free to take them or leave them.

**Comment by edmofro** [comment]
@IgorNadj when you're back next week let me know if any of those changes need re-review. Because of the force pushes I'm not sure exactly which commits to look at for the differences

**Comment by IgorNadj** [comment]
> @IgorNadj when you're back next week let me know if any of those changes need re-review. Because of the force pushes I'm not sure exactly which commits to look at for the differences

Yep no worries, it's just rebases, no actual changes. If something came back from testing that I needed to fix I would add a new commit and request re-review.

**Pull request merged**